pkg/cri/server/container_stats_list.go

```diff
@@ -65,12 +65,14 @@ func (c *criService) toCRIContainerStats(
 			return nil, fmt.Errorf("failed to decode container metrics for %q: %w", cntr.ID, err)
 		}
 
-		// this is a calculated value and should be computed for all OSes
-		nanoUsage, err := c.getUsageNanoCores(cntr.Metadata.ID, false, cs.Cpu.UsageCoreNanoSeconds.Value, time.Unix(0, cs.Cpu.Timestamp))
-		if err != nil {
-			return nil, fmt.Errorf("failed to get usage nano cores, containerID: %s: %w", cntr.Metadata.ID, err)
+		if cs.Cpu != nil && cs.Cpu.UsageCoreNanoSeconds != nil {
```

Comment:
Should be able to use the chaining from the generated code to check this. e.g. 
```go 
if cs.GetCpu().GetUsageCoreNanoSeconds() != nil 
// Or alternatively to just check the value directly
cs.GetCpu().GetUsageCoreNanoSeconds().GetValue()
```

Reply:
I'm not sure it would be clearer tbh.

The latter has the downside of changing semantics (nil vs 0 value)

The former I usually tend to avoid by default mostly because it makes nil handling implicit 😅 - e.g in a refactor like the one that led to this it would be harder to spot the new potential error condition if someone switched back to direct member access.

Can change it if needed though (or if there's a more containerd conventional guard for preconditions, but I've not come across one when debugging in the past)

Reply:
The latter likely doesn't make sense here agreed, but I do tend to prefer the former. I'm not sure if we have a hard convention but the chaining is used a bunch for [setting up options](https://github.com/containerd/containerd/blob/main/pkg/cri/server/container_create_linux.go#L175-L265) in the CRI plugin.

 It's perfectly fine as is, was just a nit 😁

Reply:
cool ✨